zeppelin-interpreter/pom.xml

```diff
@@ -44,12 +44,137 @@
     <maven.aeither.provider.version>3.0.3</maven.aeither.provider.version>
     <wagon.version>1.0</wagon.version>
     <jline.version>2.12.1</jline.version>
+    <atomix.version>3.0.0-rc4</atomix.version>
+    <atomix.netty.version>4.1.11.Final</atomix.netty.version>
+    <commons-math3.version>3.1.1</commons-math3.version>
+    <guava.version>20.0</guava.version>
+    <commons-lang3.version>3.4</commons-lang3.version>
 
     <!--plugin versions-->
     <plugin.shade.version>2.3</plugin.shade.version>
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>io.atomix</groupId>
+      <artifactId>atomix</artifactId>
+      <version>${atomix.version}</version>
+      <exclusions>
+        <exclusion>
+          <groupId>io.netty</groupId>
```

Comment:
Why not use the netty version of io.atomic and instead use another version of netty ?

Reply:
If don't modify the dependencies, compiling the zengine module will fail, I won't know how to fix it.

[INFO] Zeppelin: Zengine .................................. FAILURE [  3.343 s]
[INFO] Zeppelin: Server 0.9.0-SNAPSHOT .................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:27 min
[INFO] Finished at: 2018-09-20T10:05:32+08:00
[INFO] ------------------------------------------------------------------------
[WARNING] The requested profile "hadoop-2.7" could not be activated because it does not exist.
[WARNING] The requested profile "yarn" could not be activated because it does not exist.
[WARNING] The requested profile "pyspark" could not be activated because it does not exist.
[WARNING] The requested profile "sparkr" could not be activated because it does not exist.
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.3.1:enforce (enforce) on project zeppelin-zengine: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
[ERROR] Failed while enforcing releasability the error(s) are [
[ERROR] Dependency convergence error for com.google.guava:guava:22.0 paths to dependency are:
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-utils:3.0.0-rc4
[ERROR]         +-com.google.guava:guava:22.0
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-com.google.guava:guava:20.0
[ERROR] , 
[ERROR] Dependency convergence error for org.apache.commons:commons-lang3:3.7 paths to dependency are:
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-utils:3.0.0-rc4
[ERROR]         +-org.apache.commons:commons-lang3:3.7
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.commons:commons-lang3:3.4
[ERROR] , 
[ERROR] Dependency convergence error for io.netty:netty-handler:4.1.27.Final paths to dependency are:
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-cluster:3.0.0-rc4
[ERROR]         +-io.netty:netty-handler:4.1.27.Final
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:spark-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-org.apache.zeppelin:zeppelin-python:0.9.0-SNAPSHOT
[ERROR]       +-io.grpc:grpc-netty:1.4.0
[ERROR]         +-io.netty:netty-codec-http2:4.1.11.Final
[ERROR]           +-io.netty:netty-handler:4.1.11.Final
[ERROR] , 
[ERROR] Dependency convergence error for io.netty:netty-transport:4.1.27.Final paths to dependency are:
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-cluster:3.0.0-rc4
[ERROR]         +-io.netty:netty-transport:4.1.27.Final
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-cluster:3.0.0-rc4
[ERROR]         +-io.netty:netty-codec:4.1.27.Final
[ERROR]           +-io.netty:netty-transport:4.1.27.Final
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-cluster:3.0.0-rc4
[ERROR]         +-io.netty:netty-handler:4.1.27.Final
[ERROR]           +-io.netty:netty-transport:4.1.27.Final
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-cluster:3.0.0-rc4
[ERROR]         +-io.netty:netty-transport-native-epoll:4.1.27.Final
[ERROR]           +-io.netty:netty-transport-native-unix-common:4.1.27.Final
[ERROR]             +-io.netty:netty-transport:4.1.27.Final
[ERROR] and
[ERROR] +-org.apache.zeppelin:zeppelin-zengine:0.9.0-SNAPSHOT
[ERROR]   +-org.apache.zeppelin:zeppelin-interpreter:0.9.0-SNAPSHOT
[ERROR]     +-io.atomix:atomix:3.0.0-rc4
[ERROR]       +-io.atomix:atomix-cluster:3.0.0-rc4
[ERROR]         +-io.netty:netty-transport-native-epoll:4.1.27.Final
[ERROR]           +-io.netty:netty-transport:4.1.27.Final
[ERROR] ]
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :zeppelin-zengine